src/components/KDateTimePicker/KDateTimePicker.spec.ts

```diff
@@ -0,0 +1,237 @@
+        width: width + '',
+      },
+    })
+
+    cy.getTestId(timepickerInput).should('exist')
+    cy.getTestId(timepickerInput).invoke('outerWidth').should('eq', width)
+
+    //
+    // Open the date time picker, click "Clear" and make sure default placeholder is shown
+    //
+    cy.getTestId(timepickerInput).click()
+    cy.get('.k-popover-content').should('be.visible')
+    cy.getTestId(clearButton).should('exist')
+    cy.getTestId(clearButton).click({ force: true, multiple: true })
+    cy.getTestId(timepickerInput).should('contain.text', placeholderText)
+  })
+
+  it('renders range date picker', () => {
+    mount(KDateTimePicker, {
+      props: {
+        mode: 'date',
+        modelValue: defaulTimeRange,
+        range: true,
+      },
+    })
+    cy.getTestId(timepickerInput).should('exist')
+    cy.getTestId(timepickerInput).click()
+    cy.get('.vc-pane-container').should('exist')
+    cy.get('.vc-pane-container').find('.vc-time-select').should('not.exist')
+  })
+
+  it('renders a single date and time picker', () => {
+    mount(KDateTimePicker, {
+      props: {
+        mode: 'dateTime',
+        modelValue: today,
+        range: false,
+      },
+    })
+    cy.getTestId(timepickerInput).should('exist')
+    cy.getTestId(timepickerInput).click()
+    cy.get('.vc-pane-container').should('exist')
+    cy.get('.vc-pane-container').find('.vc-time-select').should('exist')
+    cy.getTestId(timepickerInput).find('.timepicker-display').should('contain.text', todayDateTimeString)
+  })
+
+  it('renders a range date and time picker', () => {
+    mount(KDateTimePicker, {
+      props: {
+        mode: 'dateTime',
+        modelValue: defaulTimeRange,
+        range: true,
+      },
+    })
+    cy.getTestId(timepickerInput).should('exist')
+    cy.getTestId(timepickerInput).click()
+    cy.get('.vc-pane-container').should('exist')
+    cy.get('.vc-pane-container').find('.vc-time-select').should('exist')
+  })
+
+  it('displays a valid date when "Submit" is clicked', () => {
+    mount(KDateTimePicker, {
+      props: {
+        mode: 'date',
+        modelValue: today,
+        range: false,
+      },
+    })
+
+    // Open the date time picker, click "Submit"
+    cy.getTestId(timepickerInput).click()
+    cy.get('.k-popover-content').should('be.visible')
+    cy.getTestId(clearButton).should('exist')
+    cy.getTestId(submitButton).should('exist')
+  })
+
+  it('disables "Apply" button when "Clear" is clicked', () => {
+    mount(KDateTimePicker, {
+      props: {
+        mode: 'date',
+        modelValue: today,
+        range: false,
+      },
+    })
+
+    // Open the date time picker, click "Clear"
+    cy.getTestId(timepickerInput).click()
+    cy.get('.k-popover-content').should('be.visible')
+    cy.getTestId(clearButton).should('exist')
+    cy.getTestId(submitButton).should('exist')
+
+    // "Apply" button is disabled after selection is cleared
+    cy.getTestId(clearButton).click({ force: true, multiple: true })
```

Comment:
I'm not sure you want force here as it will click even if disabled.

Reply:
I have literally spent HOURS on this issue already.

https://github.com/cypress-io/cypress/issues/4856
+
https://kongstrong.slack.com/archives/C03SXBEPFE3/p1664553325500049

I feel like it's a Cypress issue, not a Mihai issue :)